samcli/local/docker/lambda_image.py

```diff
@@ -197,7 +198,15 @@ def _build_image(self, base_image, docker_tag, layers, is_debug_go, stream=None)
             for layer in layers:
                 tar_paths[layer.codeuri] = "/" + layer.name
 
-            with create_tarball(tar_paths) as tarballfile:
+            # Set tar file permission to 777
+            # This is need for systems without unix like permission bits(Windows) while creating a unix image
+            def set_item_permission(tar_info):
+                tar_info.mode = 777
```

Comment:
This sets the permission on the host system and for every single file? If so, we should find another way to deal with this. 

Do we really need 777 on all files? Can we not scope this at all?

Reply:
This sets permission for all the files in the tarball. File permission on the host is not changed.